src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -6196,11 +6170,10 @@ private TypeSyntax ParseTypeArgument()
             // or a,... which  means this case should happen less frequently than what we're 
             // trying to solve here so we err on the side of better error messages
             // for the majority of cases.
-            SyntaxKind nextTokenKind = SyntaxKind.None;
 
             if (result.IsMissing &&
                 (this.CurrentToken.Kind != SyntaxKind.CommaToken && this.CurrentToken.Kind != SyntaxKind.GreaterThanToken) &&
-                ((nextTokenKind = this.PeekToken(1).Kind) == SyntaxKind.CommaToken || nextTokenKind == SyntaxKind.GreaterThanToken))
+                (this.PeekToken(1).Kind is SyntaxKind.CommaToken or SyntaxKind.GreaterThanToken))
```

Comment:
this is *so* much cleaner.  locals being assigned halfway through an expr to read later is not good :)